### PR TITLE
Fix SummarizeGradient

### PR DESCRIPTION
**Pull request by emailweixu**
Sometimes (perhaps caused by torch 2.1?) the cloned tensor is not a leaf. And torch does not allow setting requires_grad for non-leaf tensor.

So changed to summarizing the gradient only if requires_grad is True (summarize_tensor_gradients() does this).

### CODE REVIEW

**Review by Haichao-Zhang — DISMISSED** [review]

File: alf/layers.py [hunk 1]
```diff
@@ -3212,30 +3212,23 @@ class SummarizeGradient(ElementwiseLayerBase):
     def __init__(self, name):
         """A layer for summarizing the gradient of the input tensor.
 
-        Summarize the gradient of the input tensor. Always first cloning the
-        input tensor and then setting ``requires_grad=True`` for the cloned
-        tensor to enable gradient calculation for summarization.
+        Summarize the gradient of the input tensor. Note that if the input tensor
+        does not requires gradient (i.e. x.requires_grad is False), gradient
+        will not be summarized. (This is because )
```
Comment (hnyu) [inline alf/layers.py]: Unfinished sentence?

**Review by Haichao-Zhang — APPROVED** [review]
